src/nodes.coffee

```diff
@@ -1362,6 +1365,9 @@ exports.Class = class Class extends Base
         @ctor = method
       else if method.isStatic and method.bound
         method.context = @name
+      else if method.bound
+        @boundMethods.push method.name
+        method.constructorName = @name
```

Comment:
Attach the class name (so that it can be passed as an argument to the runtime check in the method body), not sure if this is already available to the `method` somehow?

Removed `method.bound = false`, as now it needs to know about its bound-ness when compiling itself